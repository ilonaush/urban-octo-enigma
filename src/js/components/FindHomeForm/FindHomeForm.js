import React, {Component} from 'react';
import PropTypes from 'prop-types';
import "./FindHomeForm.styl";
import Input from "../Input/Input";

export class FindHomeForm extends Component {

    constructor(props) {
        super(props);
        this.state = {
            catID: '',
        };
    }


    handleSelect = ({target = null}) => {
        this.setState({
            catID: target.value
        })
    };

    render() {
        const {onSubmit, cats} = this.props;
        const {catID} = this.state;
        return (
            <form className='fireWorkerForm' onSubmit={(e) => onSubmit(e, catID)}>
                <select onChange={this.handleSelect}>
                    <option value=" ">Choose the meowy</option>
                    {cats.map((cat) =>
                        <option key={cat.id} value={cat.id} >{cat.name}, {cat.age}</option>
                    )}
                </select>
                <Input placeholder='Location of future home'/>
                <textarea placeholder='Describe the family please'/>
                <button id='submit-btn' type='submit'>Find home for the cat</button>
            </form>
        );
    }
}

FindHomeForm.propTypes = {
    cats: PropTypes.array,
    onSubmit: PropTypes.func
};

FindHomeForm.defaultProps = {
    cats: [],
};

export default FindHomeForm;
