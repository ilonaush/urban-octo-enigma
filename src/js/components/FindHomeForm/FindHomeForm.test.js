import React from 'react';
import FindHomeForm from './FindHomeForm';
import {shallow} from "enzyme/build";


describe('FindHomeForm', function () {

    beforeEach(() => {
        this.wrapper =  shallow(
            <FindHomeForm/>
        );
    });
    it('renders correctly', () => {
        expect(this.wrapper).toMatchSnapshot();
    });

    it('changes its state on select', () => {
        expect(this.wrapper.state('cat')).toEqual(null);
        this.wrapper.find('select[name="id"]').simulate('change', {target: { value : '123', name: 'id'}});
        expect(this.wrapper.state('cat').id).toEqual('123');
    });
});
