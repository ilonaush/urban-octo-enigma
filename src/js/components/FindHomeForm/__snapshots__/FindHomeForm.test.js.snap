// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindHomeForm renders correctly 1`] = `
<form
  className="fireWorkerForm"
  onSubmit={[Function]}
>
  <select
    onChange={[Function]}
  >
    <option
      value=" "
    >
      Choose the meowy
    </option>
  </select>
  <input
    placeholder="Location of future home"
  />
  <textarea
    placeholder="Describe the family please"
  />
  <button
    id="submit-btn"
    type="submit"
  >
    Find home for the cat
  </button>
</form>
`;
